src/toolbox/shamir_secrets.rs

```diff
@@ -103,7 +230,7 @@ impl SecretShare {
             }
 
             // The From trait on Scalar only supports unsigned numerical types, so we have to kinda hack this
-            let c_i = match c_i < 0.0 { 
+            let c_i = match c_i < 0.0 {
```

Comment:
I thought you were gonna change this too, remove this part, and change the computation above into operations on Scalars to avoid the problems you described. (Isn't this somehow the same as evaluate_lagrange()? Can we avoid code duplication?

Reply:
:man_facepalming: they're variations on the same algorithm; I could rewrite this one to use Scalars or just re-use `evaluate_lagrange()`.  Probably easier to read if we go the latter, though there are some superfluous subtraction operations for the f(0) case.  I'll do that.

Reply:
I'm fine with either, I haven't looked at it in enough detail, if it's different enough I'm fine with just rewriting this.